sample_code/ep164/rev04/t.py

```diff
@@ -0,0 +1,51 @@
+from typing import Union
+from typing import overload
+from typing import Protocol
+from typing import Optional
+from typing import NamedTuple
+
+
+class Indexable(Protocol):
+    def __index__(self) -> int: ...
+
+
+class _TreeNode(Protocol):
+    left: Optional['_TreeNode']
+
+    # @property
+    # def left(self) -> Optional['_TreeNode']: ...
+    @property
+    def right(self) -> Optional['_TreeNode']: ...
```

Comment:
Using literal string for the type here `Optional['_TreeNode']` instead of `Optional[_TreeNode]`, in order for the example to run successfully. [video timestamp for context](https://youtu.be/QjFChmQHJxk?t=496)